src/components/color_generator/index.js

```diff
@@ -6,49 +6,76 @@ class ColorGenerator extends Component {
     this.state = {
       id: 0,
       palette_name: '',
-      color1: {hex: '', lockType: 'lock-open'},
-      color2: {hex: '', lockType: 'lock-open'},
-      color3: {hex: '', lockType: 'lock-open'},
-      color4: {hex: '', lockType: 'lock-open'},
-      color5: {hex: '', lockType: 'lock-open'}
+      colors: [{}, {}, {}, {}, {}]
     }
   }
 
-  handleName = (e) => {
-    const { name, value } = e.target;
+  componentDidMount() {
+    this.generateColors()
+  }
 
-    this.setState({ [name]: value })
+  generateColors = () => {
+    const colors = this.state.colors.map(color => {
+      return {hex: this.generateHex(), locked: false};
+    })
+    this.setState({colors});
   }
 
-  handleColor = (e) => {
-    const { name, value } = e.target;
+  generateHex = () => {
+    const characters = "0123456789ABCDEF";
+    let hex = '';
+    for (let i = 0; i< 6; i++) {
+      hex += characters[(Math.floor(Math.random() * 16))];
+    }
+    return hex;
+  }
 
-    let color = this.state[name];
-    color.hex = value;
+  handleName = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value })
+  }
 
-    this.setState({ [name]: color });
+  updateColor = (e) => {
+    const newHex = e.target.value;
+    const i = e.target.closest('.color-field').getAttribute('data-key');
+    let state = {...this.state};
+    state.colors[i].hex = newHex;
+    this.setState(state);
   }
 
   handleLock = (e) => {
-    const { id } = e.target;
-    let color = this.state[id];
-    console.log(id, color)
-    // const lock = 'lock';
-    // const unlock = 'lock-open';
-
-    if (color.lockType === 'lock') {
-      color.lockType = 'lock-open';
-    } else {
-      color.lockType = 'lock'
-    }
-
-    this.setState({ [id]: color })
+    const i = e.target.closest('.color-field').getAttribute('data-key');
+    const { colors } = this.state;
+    colors[i].locked = !colors[i].locked;
+    this.setState({ colors })
   }
 
   render() {
-    let {palette_name, color1, color2, color3, color4, color5} = this.state;
+    const {palette_name, colors} = this.state;
+    const colorFields = colors.map((colorObj, index) => {
```

Comment:
Could you explain what this map is doing next time we talk?

Reply:
@ehk9000 Essentially it's shortening up the code that I had written, allowing for each of the fieldsets to be dynamic.